src/store/taskList/taskListSaga.ts

```diff
@@ -0,0 +1,89 @@
+import { put, takeEvery, select } from "redux-saga/effects";
+import { PayloadAction } from "@reduxjs/toolkit";
+import { IDeleteAndTogglePayload, IEditTaskPayload, ITaskObj } from "type";
+import { TODO_ACTIONS_TYPES } from "utils/constants/actionTypes";
+import {
+  ADD_TASK_STARTED,
+  ADD_TASK_COMPLETED,
+  ADD_TASK_REJECTED,
+  EDIT_TASK_STARTED,
+  EDIT_TASK_COMPLETED,
+  EDIT_TASK_REJECTED,
+  DELETE_TASK_STARTED,
+  DELETE_TASK_COMPLETED,
+  DELETE_TASK_REJECTED,
+  SET_TASK_STATUS_STARTED,
+  SET_TASK_STATUS_COMPLETED,
+  SET_TASK_STATUS_REJECTED,
+} from "./taskListSlice";
+
+function* addTask({ payload }: PayloadAction<ITaskObj>) {
+  try {
+    yield put(ADD_TASK_STARTED());
```

Comment:
We don't need to add started here

Reply:
So then how would we be setting loading state to true before we do some task operation.
We prepare our ready to set new array inside generator function after setting loading state true. We then distpatch the completed action if our operation was a success... 
One way would be to make a generic action that "START_ACTION" that would set isLoading true